api/app/src/routes/medical/document.ts

```diff
@@ -0,0 +1,56 @@
+import { Request, Response } from "express";
+import Router from "express-promise-router";
+import { asyncHandler, getCxIdOrFail, getPatientIdFromQueryOrFail } from "../util";
+const router = Router();
+import status from "http-status";
+import { getPatient } from "../../command/medical/patient/get-patient";
+
+/** ---------------------------------------------------------------------------
+ * GET /document
+ *
+ * Queries for all available document metadata for the specified patient across HIEs.
+ *
+ * @param   req.query.patientId Patient ID for which to retrieve document metadata.
+ * @return  {DocumentRef[]}     The available documents.
+ */
+router.get(
+  "/",
+  asyncHandler(async (req: Request, res: Response) => {
+    const cxId = getCxIdOrFail(req);
+    const patientId = getPatientIdFromQueryOrFail(req);
+
+    const patient = await getPatient({ id: patientId, cxId });
+    console.log(patient);
+    // TODO: #374 will implement
+    // CommonWell.queryDocuments()
+
+    return res.status(status.OK).json([]);
+  })
+);
+
+/** ---------------------------------------------------------------------------
+ * GET /document
+ *
+ * Downloads the specified document for the specified patient.
+ *
+ * @param   req.query.patientId Patient ID for which to retrieve document metadata.
+ * @return  {docUrl: string}     The available documents.
+ */
+router.get(
+  "/download",
```

Comment:
`GET /documents` will return the list of documents, each with a download link, which should refer to this endpoint `/document/download` so we can validate access.

This means clients will be accessing this endpoint for a specific document, so I think we'll want to add the document ID as part of the URI: `/document/download/:id`

But NABD if we leave for later, unless this will impact some downstream PR relying on the signature of this endpoing.

Reply:
Good point - added this to the scope of https://github.com/metriport/metriport-internal/issues/377